### PR TITLE
feat(nx-dev): fetch plugin stats in CI

### PR DESCRIPTION
**Pull request by isaacplmann**
- Fetch plugin stats when Vercel deploys the nx-dev site.
- The nx-dev deploy will not overwrite `quality-indicators.json` if npm stats or github stars apis are down
- Adds about 1.5 minutes to the deploy time
- Consolidates all the network calls to fetch github stars into a single network call.  (Couldn't find a graphql endpoint for npm to do the same thing for those network calls)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,7 @@ jest.debug.config.js
 /graph/client/src/assets/generated-task-graphs
 /nx-dev/nx-dev/public/documentation
 /nx-dev/nx-dev/public/images/open-graph
+/nx-dev/nx-dev/pages/extending-nx/quality-indicators.json
```
Comment (meeroslav) [inline .gitignore:22]: I would leave those in for local testing

File: nx-dev/nx-dev/pages/extending-nx/registry.tsx [hunk 2]
```diff
@@ -13,9 +13,12 @@ import { useRef } from 'react';
 import { menusApi } from '../../lib/menus.api';
 import { useNavToggle } from '../../lib/navigation-toggle.effect';
 import { nxPackagesApi } from '../../lib/packages.api';
-import * as qualityIndicators from './quality-indicators.json';
 
 declare const fetch: any;
+let qualityIndicators = {};
+try {
+  qualityIndicators = require('./quality-indicators.json');
+} catch (ex) {}
```
Comment (meeroslav) [inline nx-dev/nx-dev/pages/extending-nx/registry.tsx:21]: We should always have that file, just have `deploy-build` update it with latest data

File: scripts/documentation/plugin-quality-indicators.ts [hunk 3]
```diff
@@ -75,45 +82,38 @@ main();
...
-    return data.downloads;
-  } catch (ex) {
-    return '';
-  }
+  const { data } = await axios.get(
+    `https://api.npmjs.org/downloads/point/${stringifyIntervalForUrl(
+      lastMonth
+    )}/${plugin.name}`
+  );
+  return data.downloads;
```
Comment (meeroslav) [inline scripts/documentation/plugin-quality-indicators.ts:116]: We probably don't want to break our build if npm is down for what ever reason. We can use stale data.

**Review by meeroslav — CHANGES_REQUESTED** [review]
I don't think we should remove the `quality-indicators.json` from repo nor break the deploy if npm/github api are unavailable.

**Review by meeroslav — APPROVED** [review]

**Review by juristr — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**